### PR TITLE
Otp service endpoint

### PR DESCRIPTION
**Pull request by Nexengineer**

### CODE REVIEW

File: src/In.ProjectEKA.DefaultHip/Link/PatientVerification.cs [hunk 1]
```diff
@@ -24,15 +24,17 @@ public PatientVerification(IConfiguration configuration, HttpClient httpClient)
         
         public async Task<OtpMessage> SendTokenFor(Session session)
         {
-            var uri = new Uri(configuration.GetConnectionString("OTPGenerationConnection"));
+            var urlConfig = configuration.GetValue<string>("OtpService:BaseUrl");
```
Comment (ganesanarun) [inline src/In.ProjectEKA.DefaultHip/Link/PatientVerification.cs]: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-3.1 Could you please check does it make sense to you use options pattern here?

File: src/In.ProjectEKA.HipService/appsettings.json [hunk 2]
```diff
@@ -7,8 +7,9 @@
   "AllowedHosts": "*",
   "ConnectionStrings": {
     "DefaultConnection":
-    "Host=db;Port=5432;Username=postgres;Password=welcome;Database=hipservice;",
-    "OTPGenerationConnection": "http://172.16.2.27:5000/otp/link",
-    "OTPVerificationConnection": "http://172.16.2.27:5000/otp/verify"
+    "Host=db;Port=5432;Username=postgres;Password=welcome;Database=hipservice;"
+  },
+  "OtpService": {
+    "BaseUrl":"http://localhost:5000"
```
Comment (ganesanarun) [inline src/In.ProjectEKA.HipService/appsettings.json]: Did we solve the problem of HIP talking to OTP through docker-compose service in remote env?

File: test/In.ProjectEKA.DefaultHipTest/Link/Patient/PatientVerificationTest.cs [hunk 3]
```diff
@@ -81,13 +87,8 @@ private async void ReturnSuccessOnOtpCreation()
...
                 BaseAddress = new Uri("http://localhost:5000/otp/link"),
             };
-            var mockConfSection = new Mock<IConfigurationSection>();
-            mockConfSection.SetupGet(m => m[It.Is<string>(s => s == "OTPGenerationConnection")])
-                .Returns("http://localhost:5000/otp/link");
-            configurationMock.Setup(x => x.GetSection(It.Is<string>(s=>s == "ConnectionStrings")))
-                .Returns(mockConfSection.Object);
             
-            var patientVerification = new PatientVerification(configurationMock.Object,httpClient);
+            var patientVerification = new PatientVerification(configurationMock,httpClient);
```
Comment (ganesanarun) [inline test/In.ProjectEKA.DefaultHipTest/Link/Patient/PatientVerificationTest.cs]:
```suggestion
            var patientVerification = new PatientVerification(configurationMock, httpClient);
```

File: test/In.ProjectEKA.DefaultHipTest/Link/Patient/PatientVerificationTest.cs [hunk 4]
```diff
@@ -44,13 +55,8 @@ private async void ReturnFailureOnOtpCreation()
...
                 BaseAddress = new Uri("http://localhost:5000/otp/link"),
             };
-            var mockConfSection = new Mock<IConfigurationSection>();
-            mockConfSection.SetupGet(m => m[It.Is<string>(s => s == "OTPGenerationConnection")])
-                .Returns("http://localhost:5000/otp/link");
-            configurationMock.Setup(x => x.GetSection(It.Is<string>(s=>s == "ConnectionStrings")))
-                .Returns(mockConfSection.Object);
 
-            var patientVerification = new PatientVerification(configurationMock.Object,httpClient);
+            var patientVerification = new PatientVerification(configurationMock,httpClient);
```
Comment (ganesanarun) [inline test/In.ProjectEKA.DefaultHipTest/Link/Patient/PatientVerificationTest.cs]:
```suggestion
            var patientVerification = new PatientVerification(configurationMock, httpClient);
```

File: test/In.ProjectEKA.DefaultHipTest/Link/Patient/PatientVerificationTest.cs [hunk 5]
```diff
@@ -19,13 +19,24 @@ namespace In.ProjectEKA.DefaultHipTest.Link.Patient
     [Collection("Patient Verification Tests")]
     public class PatientVerificationTest
     {
-        private readonly Mock<IConfiguration> configurationMock = new Mock<IConfiguration>();
+        private readonly IConfiguration configurationMock;
```
Comment (ganesanarun) [inline test/In.ProjectEKA.DefaultHipTest/Link/Patient/PatientVerificationTest.cs]:
```suggestion
        private readonly IConfiguration configuration;
```

File: src/In.ProjectEKA.HipService/appsettings.json (hunk 2 above)
Comment (Nexengineer) [reply src/In.ProjectEKA.HipService/appsettings.json]: No, I tried searching for env variable unable to find any

File: src/In.ProjectEKA.DefaultHip/Link/PatientVerification.cs (hunk 1 above)
Comment (Nexengineer) [reply src/In.ProjectEKA.DefaultHip/Link/PatientVerification.cs]: I added it earlier using a class but was not able to test it.

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/In.ProjectEKA.DefaultHip/Link/OtpServiceConfiguration.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4872120&bid=16151515)

File: test/In.ProjectEKA.DefaultHipTest/Link/Patient/PatientVerificationTest.cs [hunk 6]
```diff
@@ -19,13 +17,18 @@ namespace In.ProjectEKA.DefaultHipTest.Link.Patient
     [Collection("Patient Verification Tests")]
     public class PatientVerificationTest
     {
-        private readonly Mock<IConfiguration> configurationMock = new Mock<IConfiguration>();
+        private readonly IOptions<OtpServiceConfiguration> otpServiceConfigurations;
+        public PatientVerificationTest()
+        {
+            var otpService = new OtpServiceConfiguration() { BaseUrl = "http://localhost:5000" };
```
Comment (ganesanarun) [inline test/In.ProjectEKA.DefaultHipTest/Link/Patient/PatientVerificationTest.cs]: Codacy Issue found: [Remove these redundant parentheses.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4872120)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/In.ProjectEKA.DefaultHip/Link/OtpServiceConfiguration.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4872120&bid=16151515)

**Pull request merged**